src/Microsoft.ServiceFabric.Services/Client/ServicePartitionResolver.cs

```diff
@@ -637,11 +639,16 @@ public static ServicePartitionResolver GetDefault()
 
                 // wait before retry
                 await Task.Delay(
-                    new TimeSpan((long)(Rand.NextDouble() * maxRetryInterval.Ticks)),
+                    Utility.GetRetryDelay(this.GetRetryJitter(maxRetryInterval), currentRetryCount++),
                     cancellationToken);
             }
         }
 
+        private TimeSpan GetRetryJitter(TimeSpan maxRetryInterval)
+        {
+            return new TimeSpan((long)(Rand.NextDouble() * maxRetryInterval.Ticks));

```

Comment:
>new TimeSpan((long)(Rand.NextDouble() * maxRetryInterval.Ticks)); [](start = 19, length = 65)

Random threadsafety.

Also, we are operating on ticks here, where as everywhere else we seem to be operating on timespan.seconds.

Reply:
Will do the thread safety here.